### DISCUSSION TITLE
JS is not working in my deployed page. Any ideas why?

### DISCUSSION THREAD

**Discussion by LucasRegalar** [community/community#112219] [Other Feature Feedback, Questions, & Ideas] [answered]
I just uploaded a Frontend-mentor project. I am using the git-pages tool to host my website. The HTML and CSS work fine, but I seem to have a problem with my JS, since my mobile Navigation is not working. 

Does somebody got an idea, what went wrong?

Repository: https://github.com/LucasRegalar/Responsive-Landing-Page
Hosted-Page: https://lucasregalar.github.io/Responsive-Landing-Page/

Thank you so much in advance!

Love
Lucas

**Comment by kuciapakrystian** [comment]
In the script.js file, I see that the querySelector() method provides access to the HTML element. in a specific place querySelector() with a CSS class (querySelector('.classname')), while in other places it will hit with an id (querySelector('#idname')). When using an element with identifiers, the getElementById() methods are used.

For example, instead of:
const navToggle = document.querySelector('.nav-toggle');

You should use:
const navToggle = document.getElementById('nav-toggle');

Similarly, instead of:
const navMenu = document.querySelector('.nav-menu');

You should use:
const navMenu = document.getElementById('nav-menu');

After correcting these code fragments your page  should work properly. Make sure you use the appropriate ids for these elements in your HTML file.

**Reply by LucasRegalar** [reply]
Thank you for your reply! 
But, I believe I am doing that already, don't I?

**Reply by kuciapakrystian** [reply]
To further diagnose the issue, let's consider a few more potential causes:

Event Listener: Ensure that the event listener for toggling the navigation menu is properly set up and triggered when necessary, especially for mobile devices.

CSS Display Property: Check if your CSS is properly toggling the display property of the navigation menu when the toggle button is clicked. Sometimes, CSS issues can prevent elements from appearing or functioning as expected.

Console Errors: Inspect the browser console for any error messages. Sometimes, errors in other parts of your JavaScript code can prevent certain functionalities from working properly.

Viewport Meta Tag: Make sure you have the viewport meta tag set up correctly in your HTML <head> section to ensure proper rendering on mobile devices. It should look something like this: <meta name="viewport" content="width=device-width, initial-scale=1.0">.

If you've already checked these aspects and the issue persists, please let me know, and I can assist you further. Additionally, if you have specific code snippets related to the navigation functionality, sharing them could help pinpoint the problem more accurately.

**Reply by LucasRegalar** [reply]
the problem was the <html> tag. I had it set up with a type of="module" which didn't work in browsers. 

But thank you very much for your time and your effort!! :) 

Greetings
Lucas

**Comment by LucasRegalar** [comment, marked as answer]
I believe I found the problem. I used "vite" to create all my files and by default they add the JS link with a type of "module" which doesn't work in browsers, but in live-servers. 

I already changed my index.html accordingly:


  <script src="main.js"></script>
  
  and commited the new version. Now I only got the problem, that the pages-hosting algorithm still uses my old index.html file. Maybe somebody got an idea how I can change that?

**Reply by LucasRegalar** [reply]
Ah it just updated it. Everything works fine now :)
